torq.q

```diff
@@ -565,6 +562,18 @@ sys:{[cmd]
 	.lg.o[`system;"executing system command: ",cmd];
 	catcherror:{[cmd;error] .lg.e[`system;"failed to execute ",cmd,": ",error];'error};
 	@[{result:system x;.lg.o[`system;"successfully executed"];result};cmd;catcherror[cmd;]]
+	/-@[{result:system x;.lg.o[`system;"successfully executed"];result};cmd;{.lg.e[`system;"failed to execute ",cmd,": ",x];'x}]
+	};
+
+// Load additional config files, check if the specific configuration environment variable exists and load the settings if it does
+loadaddcnfg:{[envvar;cnfgpath]
+	evstring:string envvar;
+	$[""~getenv[envvar];
+	.lg.o[`fileload;"environment variable ",evstring," not set, not loading app specific config"];
+	(set[cnfgpath;getenv[envvar],"/settings/"];
```

Comment:
why is this needed? I think it's probably simpler/cleaner to not pass this `cnfgpath` variable to the function and instead just _return_ the path from the function, then at invocation we can do something like

```q
.proc.appconfig:.proc.loadaddconfig`KDBAPPCODE;
```

Reply:
this is a clever work around, I purely had it as it was the only way I saw to ensure the variable still got set, although with the change to an `if` block with an escape then this variable can hopefully (still need to check) can be called in `.proc.loadconfig` would this change things in terms of simplicity/tidiness